Lib/email/headerregistry.py

```diff
@@ -141,10 +141,8 @@ def __str__(self):
         if self.display_name is None and len(self.addresses)==1:
             return str(self.addresses[0])
         disp = self.display_name
-        if disp is not None:
-            nameset = set(disp)
-            if len(nameset) > len(nameset-parser.SPECIALS):
-                disp = parser.quote_string(disp)
+        if disp is not None and set(disp) & parser.SPECIALS:
```

Comment:
This can be written as
```python
if disp is not None and not parser.SPECIALS.isdisjoint(disp):
```

Reply:
There is also a similar code at line 90.

Reply:
Updated the code to use ``set.intersection``, which I think reads better than "not isdisjoint". I spotted a 3rd similar comparison and changed that one as well.

Reply:
`set.intersection()` creates a new set, while `set.isdisjoint()` returns a boolean and stops fast if a common item is found.